sql/core/src/main/scala/org/apache/spark/sql/GroupedData.scala

```diff
@@ -149,28 +157,70 @@ class GroupedData protected[sql](df: DataFrameImpl, groupingExprs: Seq[Expressio
    * The resulting [[DataFrame]] will also contain the grouping columns.
    */
   def mean(): DataFrame = aggregateNumericColumns(Average)
+ 
+  /**
+   * Compute the average value for given numeric columns for each group. This is an alias for `avg`.
+   * The resulting [[DataFrame]] will also contain the grouping columns.
+   */
+  def mean(colName: String, colNames: String*): DataFrame = {
+    aggregateNumericColumns(colName, colNames:_*)(Average)
+  }
 
   /**
    * Compute the max value for each numeric columns for each group.
    * The resulting [[DataFrame]] will also contain the grouping columns.
    */
   def max(): DataFrame = aggregateNumericColumns(Max)
+ 
+  /**
+   * Compute the max value for given numeric columns for each group.
+   * The resulting [[DataFrame]] will also contain the grouping columns.
+   */
+  def max(colName: String, colNames: String*): DataFrame = {
```

Comment:
seems to me you can combine the two max functions, and then run aggregate all on columns if the vararg is empty. 

also make sure you annotate the varargs